ophyd/signal.py

```diff
@@ -839,13 +843,33 @@ def get(self, *, as_string=None, connection_timeout=1.0, form='time',
             as_string = self._string
 
         self.wait_for_connection(timeout=connection_timeout)
-        info = self._read_pv.get_with_metadata(as_string=as_string, form=form,
-                                               **kwargs)
+
+        # if no timeout given, specify here:
+        timeout = kwargs.get('timeout', None)
+        if timeout is None:
+            # `timeout` not given, define here, considering `count`
+            timeout = DEFAULT_TIMEOUT
+            count = max(1, kwargs.get('count', 1))
```

Comment:
This is not going to do what is expected for arrays.  Instead of defaulting to 1 we should default to the "native size" of the PV.